zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java

```diff
@@ -1488,8 +1488,15 @@ private void runAllParagraphs(NotebookSocket conn,
             new TypeToken<List<Map<String, Object>>>() {
             }.getType());
 
-    getNotebookService().runAllParagraphs(noteId, paragraphs, getServiceContext(fromMessage),
-        new WebSocketServiceCallback<Paragraph>(conn));
+    if (!getNotebookService().runAllParagraphs(noteId, paragraphs, getServiceContext(fromMessage),
+        new WebSocketServiceCallback<Paragraph>(conn))) {
+      Note note = getNotebookService().getNote(noteId, getServiceContext(fromMessage), new SimpleServiceCallback());
```

Comment:
Could you add one comment to explain why we need to broadcast paragraphs to client ?

Reply:
Client sets all paragraph states to "pending". Now, if we stop running paragraphs when one fails, paragraphs after the failed one are still in state "pending" in client. Broadcasting them to client resets the states.

Reply:
I mean to add comment in the code so that user can understand what this piece of code is doing.

Reply:
Done.